realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -385,12 +392,18 @@ private boolean categorizeField(Element element) {
             if (!categorizePrimaryKeyField(field)) { return false; }
         }
 
-        // Check @LinkingObjects last since it is not allowed to be either @Index, @Required or @PrimaryKey
+        // @LinkingObjects cannot be @PrimaryKey or @Index.
         if (field.getAnnotation(LinkingObjects.class) != null) {
+            // Do not add backlinks to fields list.
             return categorizeBacklinkField(field);
         }
 
-        // Standard field that appear valid (more fine grained checks might fail later).
+        // Similarly, a RealmInteger cannot be a @PrimaryKey, @Index or @LinkingObject.
+        if (Utils.isRealmInteger(field)) {
+            if (!categorizeMutableRealmIntegerField(field)) { return false; }
```

Comment:
To avoid returning when `categorizeMutableRealmIntegerField(field)` returns `true`?